File: Java/Problem 2.java
public class LRUCache {

    private final int capacidad;
    private final HashMap<Integer, Integer> cache;
    private final LinkedList<Integer> ordenDeUso;

    public LRUCache(int capacity) {
        this.capacidad = capacity;
        this.cache = new HashMap<>();
        this.ordenDeUso = new LinkedList<>();
    }

    public int get(int key) {
        if (!cache.containsKey(key)) {
            return -1;
        }
        ordenDeUso.remove((Integer) key);
        ordenDeUso.addFirst(key);
        return cache.get(key);
    }

    public void put(int key, int value) {
        if (cache.containsKey(key)) {
            ordenDeUso.remove((Integer) key);
        } else if (cache.size() == capacidad) {
            int LRU = ordenDeUso.removeLast();
            cache.remove(LRU);
        }
        ordenDeUso.addFirst(key);
        cache.put(key, value);
    }

}

File: Python/Problem 2.py
class LRUCache(object):

    def __init__(self,capacity):
        self.capacidad = capacity
        self.cache = {}
        self.ordenDeUso = []

    def get(self,key):
        if key not in self.cache:
            return -1
        self.ordenDeUso.remove(key)
        self.ordenDeUso.insert(0,key)
        return self.cache.get(key)

    def put(self,key,value):
        if key in self.cache:
            self.ordenDeUso.remove(key)
        elif len(self.cache)==self.capacidad:
            LRU = self.ordenDeUso.pop()
            del self.cache(LRU)
        self.ordenDeUso.insert(0,key)
        self.cache[key]= value
